### DISCUSSION TITLE
How to record YouTube 4k stream?

### DISCUSSION THREAD

**Discussion by DarthApron** [streamlink/streamlink#4250] [Q&A] [answered]
Hi all,

when I want to record an UHD (4k) livestream from YouTube, it only detects 1080p as best quality, but not 2160p:

`C:\Users\john>streamlink https://www.youtube.com/watch?v=rw7bxTqkYYs`
`[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=rw7bxTqkYYs`
`Available streams: 144p (worst), 240p, 360p, 480p, 720p, 1080p (best)`

I found this issue from 2019: https://github.com/streamlink/streamlink/issues/2373

But we have 2021 and things may have changed.
Am I doing something wrong or why is it still not possible to record 4k livestreams?

Thanks a lot for your help.

**Comment by mkbloke** [comment, marked as answer]
> But we have 2021 and things may have changed.

Unfortunately not.

> why is it still not possible to record 4k livestreams?

The plugin as it is only supports HLS streams.  Ideally it would support YouTube's DASH implementation which would allow for the higher resolutions as well, along with features such as rewind, but it is necessary for somebody to implement this and so far, nobody has done so yet.  That's probably not the answer you want, but I'm afraid that's how it is.

**Reply by DarthApron** [reply]
You are right that I'd prefer it would work.
But I totally appreciate this proper explanation, which is a rare gem in modern Internet.
Thank you very much.
